src/framing.c

```diff
@@ -1600,7 +1616,7 @@ void test_pack(const int *pl, const int **headers, int byteskip,
               ogg_stream_packetout(&os_de,&op_de); /* just catching them all */
```

Comment:
> ``memset`` is correct. It will do the right thing.

I looked this up and I'm starting to believe it isn't. C11 6.2.6.1.6 says that when a value is stored in a structure, the padding bytes take unspecified values. If memset() writes 0s to the padding bytes in ogg_packet, then ogg_stream_packetout() modifies ogg_packet, the padding bytes become unspecified again, which could trick memcmp() into believing two semantically identical ogg_packets are different because of differences in the padding bytes.

I can't confirm this with the ogg_packet struct itself (it's probably too small,) but I've definitely seen Clang optimise struct copies by synthesising calls to memcpy(). It's allowed to do this, even with structs that have padding, because of C11 6.2.6.1.6. If it does this optimisation with an ogg_packet, it will clobber the padding bytes in the destination.